portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -41,6 +41,30 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     response.getWriter().println(json);
   }
 
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // If the user sends another POST request after the game is over, then start a new game.
+
+    // Get the input from the form.
+    String msg = getMessage(request);
+    String yes = getYes(request);
+    String email = getEmail(request);
+    response.getWriter().println("Thanks for reaching out!");
+
+    if (yes && email == "Email address") {
```

Comment:
it's a little odd to check against this magic string. I would have expected to check against the empty string. Perhaps the input change below may make it possible to change this check.